### PR TITLE
2nd & 3rd homework

### PR DESCRIPTION
**Pull request by makacies**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (demns) [inline .vscode/launch.json:1]: usually `.vscode` and any other ide settings are not commited

File: api/swagger/swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+      parameters:
+        - in: body
+          name: body
+          description: Post object that needs to be added to the store
+          required: true
+          schema:
+            $ref: '#/definitions/Post'
+      responses:
+        201:
+          $ref: "#/responses/creted"
```
Comment (demns) [inline api/swagger/swagger.yaml]:
```suggestion
          $ref: "#/responses/created"  
```

File: api/swagger/swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+        400:
+          $ref: '#/responses/badRequest'
+        403:
+          $ref: '#/responses/forbidden'
+        404:
+          $ref: '#/responses/notFound'
+  /posts/{id}:        
+    get:
+      tags:
+        - posts
```
Comment (demns) [inline api/swagger/swagger.yaml]: why do you need a tag? should it be posts or `post`?

File: api/swagger/swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+          $ref: '#/responses/forbidden'
+  /user/{username}:
+    get:
+      tags:
+        - user
+      summary: Get user by user name
+      produces:
+        - application/json
+      parameters:
+      - name: username
```
Comment (demns) [inline api/swagger/swagger.yaml]: why not id here?

File: api/swagger/swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+        type: string
+  Post:
+    type: object
+    required:
+      - id
+      - createdAt
+      - postBody
+    properties:
+      id:
+        type: integer
```
Comment (demns) [inline api/swagger/swagger.yaml]: so it's not uuid? https://github.com/makacies/nodejs-practice/pull/1/files#diff-a9245a482d8363d9f4d3a821f1beb9d4R201

File: api/swagger/swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+        example: doggie
+  Comment:
+    type: object
+    required:
+      - body
+      - createdAt
+    properties:
+      body:
+        type: string
+        format: I'm comment
```
Comment (demns) [inline api/swagger/swagger.yaml]: well ok.

File: src/routers/stats.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+router
+    .route('/')
+    .get(function (req, res) {
+        res.send(stats)
+    });
+
+function getSingleArrayOfComments(comments) {
+    let singleArray = [];
+    comments.map(item => {
```
Comment (demns) [inline src/routers/stats.js]: reduce?

File: src/statistic/calculator.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+const DAY_MS = 8.64e+7;
```
Comment (demns) [inline src/statistic/calculator.js]: seems like magic, but actually why not using 24*1000*1000*60 etc. string?

File: src/routers/auth.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+const express = require('express');
+const router = express.Router();
+const CookieName = "blogCookie";
+
+router.use(function timeLog(req, res, next) {
+    next()
```
Comment (demns) [inline src/routers/auth.js]: ?) why?)

File: .vscode/launch.json (hunk 1 above)
Comment (makacies) [reply .vscode/launch.json:1]: yeah, I know that. but this repository is not interesting to anyone, but me, so I decided to do it for my convenience.

File: api/swagger/swagger.yaml (hunk 3 above)
Comment (makacies) [reply api/swagger/swagger.yaml]: just for grouping purposes. it could be 'post' by the way

File: api/swagger/swagger.yaml (hunk 4 above)
Comment (makacies) [reply api/swagger/swagger.yaml]: It's an open question :) I'm choosing between username and id. But probably id would be more convenient.

File: api/swagger/swagger.yaml (hunk 5 above)
Comment (makacies) [reply api/swagger/swagger.yaml]: yes, it should be uuid, I forgot to update it, I'll fix it

File: src/routers/stats.js (hunk 7 above)
Comment (makacies) [reply src/routers/stats.js]: thanks, I'll try that

File: src/statistic/calculator.js (hunk 8 above)
Comment (makacies) [reply src/statistic/calculator.js]: Agreed, I'll update it.

File: src/routers/auth.js (hunk 9 above)
Comment (makacies) [reply src/routers/auth.js]: It's a relic. I've got to update it, thanks )

**Pull request merged**